connector/keystone/keystone.go

```diff
@@ -0,0 +1,158 @@
+// Package keystone provides authentication strategy using Keystone.
+package keystone
+
+import (
+	"context"
+	"fmt"
+	"github.com/dexidp/dex/connector"
+	"github.com/sirupsen/logrus"
+	"encoding/json"
+	"net/http"
+	"bytes"
+	"io/ioutil"
+)
+
+var (
+	_ connector.PasswordConnector = &keystoneConnector{}
+  	_ connector.RefreshConnector = &keystoneConnector{}
+)
+
+// Open returns an authentication strategy using Keystone.
+func (c *Config) Open(id string, logger logrus.FieldLogger) (connector.Connector, error) {
+	return &keystoneConnector{c.Domain, c.KeystoneHost,
+	c.KeystoneUsername, c.KeystonePassword, logger}, nil
+}
+
+func (p keystoneConnector) Close() error { return nil }
+
+func (p keystoneConnector) Login(ctx context.Context, s connector.Scopes, username, password string) (
+		identity connector.Identity, validPassword bool, err error) {
+	response, err := p.getTokenResponse(username, password)
```

Comment:
It's better to thread the `ctx` parameter through here, and use it when making the request to Keystone -- see [the github connector for example](https://github.com/dexidp/dex/blob/4329f407dcfefad56aa81b4d3da1a8e487bd023e/connector/github/github.go#L483-L488).